### PR TITLE
Fix Recursive URL Loader

### PR DESCRIPTION
**Pull request by hinthornw**
Previously we wouldn't follow paths like
```
relative/to/current
```
or
```
https://www.samedomain.com/i/am/relative/path
```

### CODE REVIEW

**Review by rlancemartin — COMMENTED** [review]
Thanks!

Yes, we need test for this.

Will bring in tests w/ - 
https://github.com/langchain-ai/langchain/pull/8502

(This PR was set to go in prior to repo re-structure, and has gotten delayed sadly.)

Let's prioritize now that the loader is broken.

File: libs/langchain/langchain/document_loaders/recursive_url_loader.py [hunk 1]
```diff
@@ -69,19 +73,30 @@ def get_child_links_recursive(
...
-        # Extract only the links that are children of the current URL
-        child_links = list(
-            {
-                link
-                for link in all_links
-                if link and link.startswith(current_path) and link != current_path
-            }
-        )
-
+        child_links = set()
```
Comment (rlancemartin) [inline libs/langchain/langchain/document_loaders/recursive_url_loader.py:76]: Still returns 0 links for this url `https://js.langchain.com/docs/modules/memory/integrations/` w/ `crawl_siblings =True`. odd. did not dig in further, but can confirm the updated loader is working at least for that particular link:

https://github.com/langchain-ai/langchain/pull/8502

**Comment by rlancemartin** [comment]
Can merge this if urgent to un-break.

Otherwise, lets prioritize getting the new loader in, which will have a test added:
https://github.com/langchain-ai/langchain/pull/8502

**Comment by hinthornw** [comment]
Ya fine waiting

**Pull request closed without merging**